hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestBucketList.java

```diff
@@ -108,6 +109,37 @@ public void listSubDir() throws OS3Exception, IOException {
   }
 
 
+  @Test
+  public void listObjectOwner() throws OS3Exception, IOException {
+
+    UserGroupInformation user1 = UserGroupInformation
+        .createUserForTesting("user1", new String[] {"user1"});
+    UserGroupInformation user2 = UserGroupInformation
+        .createUserForTesting("user2", new String[] {"user2"});
+
+    BucketEndpoint getBucket = new BucketEndpoint();
+    OzoneClient client = new OzoneClientStub();
+    client.getObjectStore().createS3Bucket("b1");
+    OzoneBucket bucket = client.getObjectStore().getS3Bucket("b1");
+
+    UserGroupInformation.setLoginUser(user1);
```

Comment:
In an actual cluster wouldn't the "logged in" user on s3 gateway be s3 gateway itself, not the user making the request? I think what we need to test that s3g is forwarding the user info it was given from the s3 client and OM is storing it correctly.

Reply:
For s3g, if you put an object, the object's owner will be the AWS "logged in" user, not the S3 Gateway user. this test only for `Ozone client`.